daemon.go

```diff
@@ -51,6 +55,15 @@ func NewDaemon(ctx context.Context, path string, opts ...libp2p.Option) (*Daemon
 
 	go d.listen()
 
+	sigc := make(chan os.Signal, 1)
+	signal.Notify(sigc, os.Interrupt, syscall.SIGTERM)
```

Comment:
is this actually useful? SIGTERM already terminates the program.

Reply:
@vyzo 
in daemon.go, i added this code:
```
	sigc := make(chan os.Signal, 1)
	signal.Notify(sigc, os.Interrupt, syscall.SIGTERM)
	go func(ln net.Listener, c chan os.Signal) {
		sig := <-c
		fmt.Printf("Caught signal %s: shutting down.", sig)
		ln.Close()
		os.Exit(0)
	}(d.listener, sigc)
```
so that the listener is properly closed (and removes the /tmp/p2pd.sock file) in these two cases:
1) when the user hits Ctrl-C  (sends SIGINT)
2) when this code:
```
//export stopD
func stopD() {
	p, _ := os.FindProcess(os.Getpid())
	p.Signal(os.Interrupt)
}
```
is called.  Basically, I was just sending a SIGINT when stopD is called so that the code in daemon.go is called.  

I am sure you have a better way to do it, but I needed some way to "gracefully" exit